src/librustc/middle/region.rs

```diff
@@ -890,7 +890,7 @@ fn resolve_local<'a, 'tcx>(visitor: &mut RegionResolutionVisitor<'a, 'tcx>,
     ///        | box P&
     fn is_binding_pat(pat: &hir::Pat) -> bool {
         match pat.node {
-            PatKind::Binding(hir::BindByRef(_), ..) => true,
+            PatKind::Binding(hir::BindByRef(_), ..) => true, // FIXME(tschottdorf): vetted and perhaps problematic
```

Comment:
Hmm, I think this is OK, but it's a subtle one. This has to do with when we will consider temporaries on the right-hand-side (that is, the initializer) to live for a longer time. So in particular if you write something like:

```rust
let (ref x, ref y) = (Foo { .. }, Bar { .. });
```

the temporary value created for the initializer (the tuple) would be extended to live until the end of the enclosing block (as opposed to being dropped after the `let` is complete).

To create an *implicit* ref, however, you must have a borrowed value on the RHS already. So that means you would be doing something like:

```
let Foo { x, .. } = &Foo { x: ..., ... };
```

in place of

```
let Foo { ref x, .. } = Foo { ... }
```

In the former case (the implicit ref version), the temporary is created by the `&` expression, and indeed its lifetime would be extended to the end of the block, but due to a different rule, not `is_binding_pat`.

It's probably worth copying this comment into the source as a note for why it's ok to just look for explicit `ref` keywords here.